.github/workflows/container.yml

```diff
@@ -53,23 +57,115 @@ jobs:
       - name: Check out the code
         uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3.0.2
 
+      - name: Set up Go
+        uses: actions/setup-go@b22fbbc2921299758641fab08929b4ac52b32923 # tag=v3.2.0
+        with:
+          go-version-file: .go-version
+          check-latest: true
+          cache: true
+
+      - name: Set up Rust
+        # Yes, oddly `rustup show` installs the toolchain (:
+        # https://github.com/rust-lang/rustup/issues/2686
+        run: rustup show
+
+      - name: Rust cache
+        uses: Swatinem/rust-cache@cb2cf0cc7c5198d3364b9630e2c3d457f160790c # tag=v1.4.0
+        with:
+          working-directory: ./bpf
+
+      - name: Install LLVM (Aya, BPF dependency)
+        run: |
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+          echo -e "deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main\n" | sudo tee /etc/apt/sources.list.d/llvm.list
+          sudo apt-get update -y
+          sudo apt-get install -yq llvm-14-dev libclang-14-dev
+
+      - name: Run eBPF toolchain setup
+        run: |
+          make -C bpf setup
+          cd bpf && cargo check
+
+      - name: Build BPF
+        run: make bpf
+
+      - uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # tag=v3.1.0
+        with:
+          name: ebpf-object-file
+          path: pkg/profiler/cpu-profiler.bpf.o
+          if-no-files-found: error
+
+  build-binaries:
+    name: Build binaries using goreleaser
+    runs-on: ubuntu-latest
+    needs: build-dependencies
+    container:
+      image: goreleaser/goreleaser-cross:v1.18.3 # ${{ env.GORELEASER_CROSS_VERSION }}
```

Comment:
The `goreleaser-cross` image bothers me a little, I wonder if there exists a generic and maintained C cross-compilation image, so we can keep using GH Action to install Go and Goreleaser and retain flexibility version-wise. That may be what you alluded to earlier